### DISCUSSION TITLE
Browser window is not maximized/minimized on pressing F11 when remote desktop is connected.

### DISCUSSION THREAD

**Discussion by a-imran99** [Ylianst/MeshCentral#5707] [Issues] [answered]
Browser window is not maximized/minimized on pressing F11 when remote desktop is connected.

When we connect to a remote system and try to maximize the browser window in local machine. The F11 is not working.

**Comment by si458** [comment]
this is because all the keypresses when you connect are sent to the remote device, so pressing f11 will just maximized the remote computers browser and not yours (only exceptions are browser shortcuts like CTRL+W to close a window)

to maximize your local browser you can press the full screen button at the top right hand side, which will make the window the remote screen full size of your browser page (not including your toolbars)
![Screenshot 2024-01-18 120432](https://github.com/Ylianst/MeshCentral/assets/765314/7790eeb1-85d1-431c-9ca0-ebb369ef683a)
OR
BEFORE you click connect press F11 to maximize your browser then click connect, then once you disconnect, press F11 again

**Reply by a-imran99** [reply]
Hi @si458, Thanks for the quick reply, I have seen this feature to maximize the browser screen.

I have not tested by sending F11 key press to remote system. will test this feature.

But when we press windows key , it is opening search on both local and remote system. the windows key is not only sent to remote device but also works in host machine

**Reply by a-imran99** [reply]
How to make F11 exception similar to CTRL+W @si458

**Reply by si458** [reply]
sadly this is a problem we have, how many different keyboard press combos are the we need to block?
do some we send and some we dont?
some browsers take control of the key presses and wont sent to the remove device FULL STOP!

sadly the exception seems to be browser controlled, so the browser makes the decision what can be sent and what cant
like CTRL+SHIFT+ESC doesnt sent and just opens the local task manager rather than on the remote device

**Reply by a-imran99** [reply]
So we can't use F11 to maximize or minimize the window, instead we need to use the icon on the top right? Is that correct? @si458

**Reply by si458** [reply]
you havent read my reply correctly...

to maximize your local browser you can press the full screen button at the top right hand side, which will make the window for the remote screen the full size of your browser page (not including your toolbars)
OR
BEFORE you click 'connect' press F11 to maximize your browser then click 'connect', then once you 'disconnect', press F11 again to come out of full screen mode

**Reply by a-imran99** [reply]
yes @si458, I got you, But we can't do the seccond one because it is not feasible with our changes, we have enabled auto connect to remote system. but we can use the icon on the top  right corner to maximize the windows thats our only option.

**Reply by si458** [reply]
the is no 'auto connect' feature on the 'Desktop' tab thats im aware of?
where are you seeing this? is it a setting you have enabled?
screenshots/video?

**Reply by a-imran99** [reply]
@si458, yes we have enabled it, I would not say settings, but modified the way itself.
I can share screen record.

**Reply by si458** [reply, marked as answer]
ah right ok no worries!
if you have modified files then yes it wont work out of the box,
unless u press F11 BEFORE you click the 'connect' button
OR
you press the full screen button to make the it the full size of your browser window (excluding the tabs, so not really FULL SCREEN)

**Reply by a-imran99** [reply]
Thanks @si458.

**Reply by a-imran99** [reply]
I have one more query regarding some other thing. 

Meshcentral doesn't work the way like screen-connect, anydesk (for mac), VNC. Is this because of mesh is a browser application and others are not?
@si458

**Reply by si458** [reply]
yes, any web browser application is different compared to there counterpart full desktop application,
you could look into the MeshCentralRouter application (which is a windows exe app) and try to automate things with that maybe? never tried to automate it tho, ive only used it instead of the web app
